### ISSUE TITLE
Resending request on HTTP interceptor

### ISSUE THREAD

**Issue by ALGdev** [microsoft/cpprestsdk#1351]
Hi,
My team and I have a server (NET Core) and client side (CppRest).
Our Authorization system is via token and the client side has to check the 401 status code on response in order to refresh the token.

Thanks to the HTTP Interceptors we are checking the status code.

Every time the client side receives an Unauthorize code we want to resend the request that produced that code with the refreshed token. Is there a way to manage this? We are trying it but when we produce the request it does not even reach the endpoint.
 
Thank you.

**Comment by garethsb-sony** [comment]
What are 'HTTP Interceptors' when discussing C++ REST SDK? (They seem to be an Angular thing?)
Are you referring to `http_pipeline_stage` support in `http_client`?

**Comment by ALGdev** [comment]
Yes, I was referring to `http_pipeline_stage `support. Sorry for the mistake.
